src/System.Text.Json/tests/Serialization/JsonElementTests.cs

```diff
@@ -145,5 +146,32 @@ public void Verify()
                 Assert.Equal("Hello", Array[3].ToString());
             }
         }
+
+        [Theory,
+            InlineData(5),
+            InlineData(10),
+            InlineData(20),
+            InlineData(1024)]
+        public void ReadJsonElementFromStream(int defaultBufferSize)
```

Comment:
Did you verify this actually causes the new code to be hit? The issue is that although the buffer size is specified, we grab from the arraypool which typically will have larger blocks (e.g. 4K).

Reply:
Yes, it does fail without this. The smallest ArrayPool bucket is 16 bytes, so what we see here is 16, 32, and 1024 in reality.